src/typechecker/FStar.TypeChecker.NBE.fst

```diff
@@ -881,13 +881,17 @@ and translate_fv (cfg: config) (bs:list t) (fvar:fv): t =
                         iapp = iapp cfg;
                         translate = translate cfg bs;
                       } in
-                      match prim_step.interpretation_nbe callbacks args' with
+                      debug (fun () -> BU.print1 "Caling primop with args = [%s]\n"
+                                    (List.map (fun (x, _) -> t_to_string x) args' |> String.concat "; "));
+                      let univs, rest = List.span (function ({nbe_t=Univ _ }, _) -> true | _ -> false) args' in
+                      let univs = List.map (function ({nbe_t=Univ u}, _) -> u | _ -> failwith "Impossible") univs in
```

Comment:
Split the univs out of the arguments and pass them separately to the interpretation